### ISSUE TITLE
Index templates drop is_write_index parameter from alias metadata

### ISSUE THREAD

**Issue by imotov** [elastic/elasticsearch#52152]
Reported on the [discuss forum](https://discuss.elastic.co/t/is-write-index-true-index-template/218598).

To reproduce:

```
PUT _template/bar_template
{
  "index_patterns": ["bar*"],
  "aliases" : {
    "bars" : {
      "is_write_index": true,
      "search_routing" : "1,2",
      "index_routing" : "2",
      "filter" : { "term" : { "user" : "kimchy" } }
    }
  }
}

GET _template/bar_template
```

Result: the `is_write_index` parameter is dropped from the template.

**Comment by gaobinlong** [comment]
@imotov , if `is_write_index` is set to `true` in a template, then the first created index using the template is the write index, but another index cannot be created unless `is_write_index` is set to `false` in the request body when creating. Is this behavior correct?

**Comment by alaled** [comment]
@gaobinlong Issue is that Logstash creates indexes like so:
output {
  elasticsearch {
    id => ""
    hosts => []
    user => ""
    password => ""
    index => "haproxy-%{+YYYY.MM.dd-HH}"
    template_name => "haproxy"
  } 
}
and if "is_write_index" parameter cannot be added to "haproxy" template I don't have a way to use ILM.

**Comment by imotov** [comment]
@gaobinlong I think that would be the most logical behaviour, but might not be the most practical. I opened this issue for awareness of the fact that this parameter is ignored. What's the best way to handle this parameter is a subject for a separate discussion in my opinion.

**Comment by gaobinlong** [comment]
OK, I got it.  @imotov , I have made a PR for this issue, can you help to reivew the code changes?

**Comment by gaobinlong** [comment]
@imotov, setting `is_write_index` to `true` in a index template maybe useless, because rollover in ILM can set the new index as the write index automatically, and we cannot set a alias with the same name of rollover alias in a template as follows:

`
PUT _template/test_template
{
  "index_patterns": [
    "test-*"
  ],
  "settings": {
    "number_of_shards": 1,
    "number_of_replicas": 0,
    "index.lifecycle.name": "test-policy",
    "index.lifecycle.rollover_alias": "testalias"
  },
  "aliases": {
    "testalias": {
      "is_write_index": true
    }
  }
}
`
the settings above will trigger a execption: java.lang.IllegalArgumentException: Rollover alias [testalias] can point to multiple indices, found duplicated alias [[testalias]] in index template [test_template].

**Comment by imotov** [comment]
Well, yes it is not very helpful. And in the use case that you have demonstrated it is indeed useless. 

The only use case for it that I can think of would be to setup two patterns. One pattern would match `foo-*` and set all settings and mappings for indices `foo-NNNNNN` . Another pattern would match only `*-000001` and only set `"is_write_index": true` for the first index in the series. This can automate bootstrapping of the ILM indices to some degree. But I am not totally sure it's worth it.

**Comment by dakrone** [comment]
I believe this was fixed by #63162 so I am going to close this for now.
